knowledge-base/custom-localization-dotnet-maui-dataform.md

```diff
@@ -26,51 +26,44 @@ ticketid: 1663273
 
 ## Description
 
-When working with the [DataForm](https://docs.telerik.com/devtools/maui/controls/dataform/overview) for .NET MAUI, you might need to customize the error messages displayed for validation errors. Specifically, you want to change the default range validation error message to a custom one. 
+When working with the [DataForm](https://docs.telerik.com/devtools/maui/controls/dataform/overview) for .NET MAUI, you might need to customize the error messages displayed for validation errors. For example, you want to change the default range validation error message or required error message to a custom one. 
 
 This KB article also answers the following questions:
 - How to customize error messages in .NET MAUI DataForm when validation is implemented through attributes in your ViewModel?
 - How to change the validation error messages in DataForm through validation attributes?
 
 ## Solution
 
-To customize the error messages in the DataForm component, create a custom localization manager by extending the `TelerikLocalizationManager` class. Override the `GetString` method to return custom messages for specific keys. 
+When you use data annotations in your ViewModel for validation, such as "Required", for example:
```

Comment:
I suggest slightly restructuring the article for better readability:


Depending on the way you implement your ViewModel, you can customize the validation error messages in two ways:

1. [By using the data annotations in your ViewModel](#using-data-annotations)
1. [By using your own localization resources](#using-your-own-localization-resources)

### Using Data Annotations

When you use data annotations in your ViewModel for validation, such as `Required`, the validation error message is provided by the attribute itself. If you do not specify your own error message explicitly, a default one is provided by the .NET framework, for example:

  ```C#
  public class MyViewModel
  {
      [Required]
      public string MyProperty { get; set; }
  }
  ```

To provide a custom message instead, add it manually:

```C#
  public class MyViewModel
  {
      [Required(ErrorMessage = "My custom error message.")]
      public string MyProperty { get; set; }
  }
  ```

### Using Your Own Localization Resources

If you have your own localization resources, you can localize the validation error message:

  ```C#
  public class MyViewModel
  {
      [Required(ErrorMessageResourceType = typeof(MyLocalizationResources),
                ErrorMessageResourceName = nameof(MyLocalizationResources.MyCustomErrorMessage))]
      public string MyProperty { get; set; }
  }
  ```

Reply:
both approaches are through data annotations, just in the first case the message is directly set, in the second- it's taken from local resources. still, the approach is the same